src/main/java/com/amazon/opendistroforelasticsearch/security/OpenDistroSecurityPlugin.java

```diff
@@ -188,9 +189,7 @@
     private final boolean sslOnly;
     private final List<String> demoCertHashes = new ArrayList<String>(3);
     private volatile OpenDistroSecurityFilter odsf;
-    private volatile Environment environment;
     private volatile IndexResolverReplacer irr;
-    private AtomicBoolean externalConfigLogged = new AtomicBoolean();
```

Comment:
Moving environment and externalConfigLogged to AuditLogImpl to consolidate audit logging changes. 
When node starts, configuration may not be loaded yet to log external configuration and this OpenDistroSecurityPlugin must be subscribed as well. To avoid doing that, moved the functionality to AuditLogImpl